### ISSUE TITLE
Change in import behavior with custom "resolve" function in 3.9.6

### ISSUE THREAD

**Issue by youngbob** [patriksimek/vm2#398]
If a custom "resolve" function is provided, it seems to be called even if the module is not in the allowed list of external modules. This is a change from previous versions.

```javascript
const { NodeVM } = require('vm2')

const vm = new NodeVM({
  require: {
    external: ['jose'],
    resolve: require.resolve,
  },
})

vm.run("var jmespath = require('jmespath')", 'vm.js')
```

In 3.9.5, this code throws `VMError [Error]: The module 'jmespath' is not whitelisted in VM.` In 3.9.6, it passes without error.

**Comment by XmiliaH** [comment]
Relative path requires did allow to require any module under `options.root` circumventing the whitelist altogether. Therefore, the whitelist check was moved to a later point in time.

**Comment by XmiliaH** [comment]
The access check is done when trying to access files through https://github.com/patriksimek/vm2/blob/532120d5cdec7da8225fc6242e154ebabc63fe4d/lib/resolver.js#L153.

**Comment by youngbob** [comment]
I think this line is the culprit:

https://github.com/patriksimek/vm2/blob/532120d5cdec7da8225fc6242e154ebabc63fe4d/lib/resolver-compat.js#L291

If the custom resolver returns a path, it gets automatically added to the "external" list. Removing that line restores the previous behavior.

**Comment by XmiliaH** [comment]
Thanks, I see the problem now.

**Comment by XmiliaH** [comment]
Would you mind to check if the behavior with https://github.com/patriksimek/vm2/pull/400 is as expected.

**Comment by youngbob** [comment]
> Would you mind to check if the behavior with #400 is as expected.

#400 fixes the issue for me. The exact error message returned is different, but that's an improvement.

**Comment by XmiliaH** [comment]
Should be fixed in v3.9.7
